account_journal_lock_date/wizards/wizard_lock_account_journal.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2017 ACSONE SA/NV
+# Copyright 2017 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models, api, _, SUPERUSER_ID
+
+from ..exceptions import JournalLockDateError
+
+
+class WizardLockAccountJournal(models.TransientModel):
+    _name = "wizard.lock.account.journal"
+    _description = "Lock Account Journal"
+
+    company_id = fields.Many2one(
+        comodel_name='res.company', string="Company", required=True,
+        default=lambda self: self.env.user.company_id)
+    journal_ids = fields.Many2many(
+        'account.journal', rel='wizard_lock_account_move_journal',
+        string='Journals', required=True)
+    lock_date = fields.Date(string='Lock Date', required=True)
+    permanent_lock = fields.Boolean('Permanent Lock')
+
+    @api.multi
+    def _check_execute_acces_rights(self):
+        has_adviser_group = self.env.user.has_group(
+            'account.group_account_manager')
+        if not (has_adviser_group or self.env.uid == SUPERUSER_ID):
+            raise JournalLockDateError(
+                _("You are not allowed to execute this action."))
+
+    @api.multi
+    def _check_execute_allowed(self):
+        self.ensure_one()
+        obj_move = self.env['account.move']
+        fy_lock_date = self.company_id.fiscalyear_lock_date
+        if fy_lock_date and self.lock_date < fy_lock_date:
+            raise JournalLockDateError(
+                _("You cannot choose a lock date lower than company "
+                  "fiscalyear lock date %s.") % fy_lock_date)
+        draft_move = obj_move.search(
+            [('state', '=', 'draft'),
+             ('journal_id', 'in', self.journal_ids.ids),
+             ('date', '<=', self.lock_date)],
+            order='date')
+        if draft_move:
+            raise JournalLockDateError(
+                _("Unposted move in period/jounal selected, "
+                  "please post it before locking them"))
+
+    @api.multi
+    def lock_moves(self):
+        obj_move = self.env['account.move']
+        # Lock moves with date lower than lock_date.
+        to_lock_moves = obj_move.search(
+            [('state', '=', 'posted'),
+             ('locked', '=', False),
+             ('journal_id', 'in', self.journal_ids.ids),
+             ('date', '<=', self.lock_date)],
+            order='date')
+        to_lock_moves.lock_move()
+        # Unlock moves with date bigger than lock_date.
```

Comment:
Was this in another module you merged? I think I remember @fclementic2c wanted to have the locked field completely irreversible.

If the `locked` field can be changed, then it probably has little added value over the lock date itself.

Reply:
@sbidoul On this PR i tried to consolidate modules that was on 10, but they were too many, see https://github.com/OCA/account-financial-tools/issues/613, this was not in the modules from 10, but as a logic since you reset the lock_date to a previous one, you are doing it to unlock the moves after that date, so you modify them. I am waiting for others feedback, and update the code...

Reply:
Yep, I'm fine with the approach. My opinion is that the `locked` field is not necessary. The permanent lock date is sufficient. But let's see what other say indeed.

Reply:
@sbidoul The locked field is used to lock account.move independently, you have a button in the form view, and you can lock it, see https://github.com/OCA/account-financial-tools/blob/10.0/account_move_locking/models/account.py